### ISSUE TITLE
`dict with` returns the dict

### ISSUE THREAD

**Issue by jrpat** [msteveb/jimtcl#241]
According to [jim's documentation](http://jim.tcl.tk/fossil/doc/trunk/Tcl_shipped.html#_dict) (and [Tcl's](https://www.tcl.tk/man/tcl8.6/TclCmd/dict.html#M27)) the return of `dict with ...` should be the result of the script, but instead the dict itself is returned.

Minimal example:

```
. set d {a 1 b 2}
a 1 b 2
. dict with d {list foo bar}
a 1 b 2
```

Jim is a really lovely project. Thanks for all your work on it!

**Comment by msteveb** [comment]
Thanks for the report (and the compliment!). I did a naive fix but it has problems with test dict-22.11 in tests/dict.test so will need a bit more thought.

**Comment by jrpat** [comment]
Thanks for the fix! 🍻
